ci/docker/centos-7-cpp.dockerfile

```diff
@@ -37,7 +41,7 @@ COPY ci/scripts/install_sccache.sh /arrow/ci/scripts/
 RUN bash /arrow/ci/scripts/install_sccache.sh unknown-linux-musl /usr/local/bin
 
 ENV PATH=/opt/cmake-${cmake}/bin:$PATH \
-    CC=/usr/bin/gcc \
-    CXX=/usr/bin/g++ \
+    CC=/opt/rh/devtoolset-8/root/usr/bin/cc \
+    CXX=/opt/rh/devtoolset-8/root/usr/bin/c++ \
     EXTRA_CMAKE_FLAGS="-DCMAKE_C_COMPILER=$CC -DCMAKE_CXX_COMPILER=$CXX" \

```

Comment:
These might not be needed anymore, I think just using the devtoolset will be enough. This was to force using the default system compilers for the gcc 4.8 builds/tests.

Reply:
It's been a few days but iirc cmake does not pick up the correct compilers even with dts sourced, I'll test it again locally though

Reply:
Works without them :+1: